homeassistant/components/generic/config_flow.py

```diff
@@ -168,10 +168,15 @@ async def async_test_still(hass, info) -> tuple[dict[str, str], str | None]:
     return {}, f"image/{fmt}"
 
 
-def slug_url(url) -> str | None:
+def slug_url(hass, url) -> str | None:
     """Convert a camera url into a string suitable for a camera name."""
     if not url:
         return None
+    if not isinstance(url, template_helper.Template):
+        url = cv.template(url)
+        url.hass = hass
+    # We shouldn't have any exceptions here, because we verified the url earlier
+    url = url.async_render(parse_result=False)
     return slugify(yarl.URL(url).host)
```

Comment:
Templates have an `is_static` boolean. That way you can test if it's a static value and if so, you can definitely get the host out.

Reply:
I ended up not needing this, but please let me know if I missed something.